### PR TITLE
Fixed some issues.

### PR DESCRIPTION
**Pull request by MiqayelChaloyan**
### Fixed some issues.

   *I have set up a profile page* <br>
   *I added a privacy policy question to the sign up page*  <br>
   *I adjusted the operation of the mod inicated in the image.* <br>
   <br>
   *I removed the unnecessary styles.* <br>
sign_up
modal

### CODE REVIEW

File: screens/ContactsUsers/index.js [hunk 1]
```diff
@@ -37,24 +38,23 @@ const MessagesUsers = ({navigation}) => {
...
   const QuickActions = qaItem => {
     return (
-      <View style={styles.qaContainer}>
-        <View style={styles.button}>
-          <Pressable
-            onPress={() => deleteItem(qaItem.id)}
-            style={styles.removedContain}>
+      <TouchableOpacity
+        onPress={() => deleteItem(qaItem.id)}
+        style={styles.removedContain}>
```
Comment (M7Arman) [inline screens/ContactsUsers/index.js:45]: rename this style

File: screens/AllUsersListScreen/index.js [hunk 2]
```diff
@@ -11,21 +12,35 @@ import RenderImagePairs from './RenderImagePairs';
...
+    const result = initialData.filter(item => {
+      let param = item.fullName.toLowerCase();
+      return param.indexOf(searchItemValue) > -1;
+    });
+    setData(result);
+  };
+
+  useEffect(() => {
+    return handleSearch();
+  }, [initialData, searchItemValue, setData]);
```
Comment (M7Arman) [inline screens/AllUsersListScreen/index.js:42]: it's not a good practice to use outside function in useEffect, you should define that function inside useEffect

Comment (M7Arman) [reply screens/AllUsersListScreen/index.js:42]: isn't there eslint warning for this?

File: screens/ImagesScreen/index.js [hunk 3]
```diff
@@ -30,7 +30,6 @@ const ImagesScreen = ({navigation}) => {
     }
     setModalVisible(false);
     setSheet('');
-    // eslint-disable-next-line react-hooks/exhaustive-deps
   }, [sheet]);
```
Comment (M7Arman) [inline screens/ImagesScreen/index.js:33]: you should fix linter issue, not turn off them ))

File: screens/authentication/signUp/SignUpScreen.jsx [hunk 4]
```diff
@@ -168,21 +175,41 @@ const SignUpScreen = ({navigation}) => {
...
+            )}
+          </View>
           <View style={styles.signUpFooter}>
-            <View style={styles.button}>
-              <TouchableOpacity
-                disabled={!isValid}
-                onPress={() => handleSubmit()}>
+            <TouchableOpacity
+              disabled={!isValid}
+              onPress={() => handleSubmit()}>
```
Comment (M7Arman) [inline screens/authentication/signUp/SignUpScreen.jsx:208]: onPress={handleSubmit}>

File: .eslintrc.js [hunk 5]
```diff
@@ -28,8 +28,9 @@ module.exports = {
     'react-native/no-raw-text': 2,
     'react-native/no-single-element-style-arrays': 2,
     'no-console': 0,
-    'react-hooks/exhaustive-deps': 2,
     disallowMultipleVarDecl: 0,
     'react/react-in-jsx-scope': 0,
+    'react-hooks/rules-of-hooks': 'error',
+    'react-hooks/exhaustive-deps': 'off',
```
[the remark this answers is not in the document]
Comment (MiqayelChaloyan) [reply .eslintrc.js:34]: Ok, i will change it.

File: components/Modal/Modal.jsx [hunk 6]
```diff
@@ -6,7 +6,7 @@ import {theme} from '../../assets/theme/theme';
 import StarIcon from '../../assets/icons/Star.svg';
 import styles from './style';
 
-const CustomModal = ({isModalVisible, setModalVisible, navigation}) => {
+const CustomModal = ({visible, navigation, onClose}) => {
```
[the remark this answers is not in the document]
Comment (MiqayelChaloyan) [reply components/Modal/Modal.jsx:9]: Ok, i will change it.

File: screens/FeedScreen/index.js [hunk 7]
```diff
@@ -29,6 +32,35 @@ const FeedScreen = ({navigation}) => {
...
+
+  const closeModal = async () => {
+    try {
+      await setDataStorage('@modalStatus', 'closed');
+      setShowModal(false);
+    } catch (error) {
+      console.error('Error storing modal status:', error);
+    }
+  };
+
```
[the remark this answers is not in the document]
Comment (MiqayelChaloyan) [reply screens/FeedScreen/index.js:63]: Ok, will add that part.

File: screens/ContactsUsers/index.js (hunk 1 above)
Comment (MiqayelChaloyan) [reply screens/ContactsUsers/index.js:45]: Ok, i will change it.

File: screens/AllUsersListScreen/index.js (hunk 2 above)
Comment (MiqayelChaloyan) [reply screens/AllUsersListScreen/index.js:42]: No, eslint didn't warn. but I removed the function from depends.

File: screens/ImagesScreen/index.js (hunk 3 above)
Comment (MiqayelChaloyan) [reply screens/ImagesScreen/index.js:33]: Ok, I will fix it )

File: screens/authentication/signUp/SignUpScreen.jsx (hunk 4 above)
Comment (MiqayelChaloyan) [reply screens/authentication/signUp/SignUpScreen.jsx:208]: Ok, i will change it.

**Pull request merged**